Infrastructure: correct an error associated with `QColor` in "Room properties" and "Room symbol" classes

SlySven opened the pull request:
In a couple of places a `QColor` in the `dlgRoomSymbol` class - which was duplicated in the new `dlgRoomProperties` one - was being compared to a `nullptr` to determine if it has been set to anything. This is incorrect because the item will have a memory address whether it has been assigned a meaningful colour value or not. Instead the `QColor::isValid()` method is the correct way to do this. (An invalid or uninitiated QColor has red, green, blue and alpha values all being 0 out of 255 - i.e. it would be transparent!)

As to whether the code now works correctly I have not been able to exhaustively check. This ought to be confirmed before it is merged.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Fix: correct an error associated with QColor in room properties and symbol

In a couple of places a `QColor` in the `dlgRoomSymbol` class - which was
duplicated in the new `dlgRoomProperties` one - was being compared to a
`nullptr` to determine if it has been set to anything. This is incorrect
because the item will have a memory address whether it has been assigned a
meaningful colour value or not. Instead the `QColor::isValid()` method is
the correct way to do this. (An invalid or uninitiated QColor has red,
green, blue and alpha values all being 0 out of 255 - i.e. it would be
transparent!)

As to whether the code now works correctly I have not been able to
exhaustively check. This ought to be confirmed before it is merged.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

Kebap commented on src/dlgRoomProperties.cpp:
```diff
@@ -420,7 +420,7 @@ QFont dlgRoomProperties::getFontForPreview(QString symbolString)
 void dlgRoomProperties::slot_openSymbolColorSelector()
 {
-    auto* dialog = (selectedSymbolColor != nullptr) ? new QColorDialog(selectedSymbolColor, this) : new QColorDialog(defaultSymbolColor(), this);
+    auto* dialog = new QColorDialog(selectedSymbolColor.isValid() ? selectedSymbolColor: defaultSymbolColor(), this);
```
```suggestion
    auto* dialog = new QColorDialog(selectedSymbolColor.isValid() ? selectedSymbolColor : defaultSymbolColor(), this);
```

Kebap commented on src/dlgRoomSymbol.cpp:
```diff
@@ -198,7 +198,7 @@ QFont dlgRoomSymbol::getFontForPreview(QString text) {
 void dlgRoomSymbol::slot_openColorSelector()
 {
-    auto* dialog = selectedColor != nullptr ? new QColorDialog(selectedColor, this) : new QColorDialog(defaultColor(), this);
+    auto* dialog = new QColorDialog(selectedColor.isValid() ? selectedColor: defaultColor(), this);
```
```suggestion
    auto* dialog = new QColorDialog(selectedColor.isValid() ? selectedColor : defaultColor(), this);
```

Kebap commented:
> As to whether the code now works correctly I have not been able to exhaustively check. This ought to be confirmed before it is merged.

Maybe also check if it worked alright before the change? 😉

SlySven committed:
Apply suggestions from code review

Fix up a couple of missing spaces.

Signed-off by: Stephen Lyons <slysven@virginmedia.com>

Co-authored-by: Kebap <kebap_spam@gmx.net>

SlySven commented:
> Maybe also check if it worked alright before the change? :wink:

All the testing I was able to perform didn't show any problems with the new code - though the prior code was not throwing up any errors that anyone else had reported or that I could detect either.

OT We should perhaps be disabling the "Reset text color" button (whether it is a `QPushButton` or a `QToolButton`) when there is no colour being applied to the symbol. However the new "Room properties" dialogue has the same defect in regards to room symbol text colours as the other dialogue - it only considers the "first" room in the selection when it considers whether there is a colour on those texts...

Kebap commented:
> the prior code was not throwing up any errors (...) that I could detect either.

I had almost assumed so..
